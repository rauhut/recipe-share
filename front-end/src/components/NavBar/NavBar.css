.search-bar {
  width: 350px !important;
}

.navbar {
  background-color: "#cc1414";
}

.navbar-brand {
  color: white !important;
  font-weight: 400;
}
