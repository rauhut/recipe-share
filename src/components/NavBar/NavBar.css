.search {
  flex-grow: 1;
  display: flex;
  align-items: center;
  max-width: 600px;
}

.search-bar {
  width: 100%;
}

.search-btn {
  margin-left: -93px;
}

.navbar {
  background-color: #cc1414;
}

.home-nav {
  flex-grow: 0 !important;
}

.navbar-brand {
  color: white !important;
  font-weight: 400;
}

.navbar-collapse {
  flex-grow: 0;
}
