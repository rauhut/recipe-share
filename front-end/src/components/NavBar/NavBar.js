import ProfileDropdown from "../Profile/ProfileDropdown";
import React from "react";
import { Link } from "react-router-dom";
import {
  Collapse,
  Navbar,
  NavbarBrand,
  Nav,
  Input,
  NavbarText,
  NavbarToggler,
} from "reactstrap";
import "./NavBar.css";

class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
      searchEntry: "",
      recipes: [],
    };
  }

  onSearchChange = (event) => {
    this.setState({ searchEntry: event.target.value });
  };

  toggle = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  render() {
    const { user } = this.props;
    const { isOpen } = this.state;

    return (
      <div>
        <Navbar light expand="md">
          <Collapse className="home-nav" isOpen={isOpen} navbar>
            <NavbarBrand href="/">Recipe Share</NavbarBrand>
          </Collapse>
          <div className="search">
            <Input
              className="search-bar"
              type="search"
              name="search"
              id="searchInput"
              placeholder="Search for a Recipe"
              onChange={this.onSearchChange}
            />
            <Link className="search-btn" to={`/search`}>
              <button
                className="button-primary"
                onClick={() =>
                  this.props.onRecipeSearch(this.state.searchEntry)
                }
              >
                Search
              </button>
            </Link>
          </div>
          <NavbarToggler onClick={this.toggle} />
          <Collapse isOpen={isOpen} navbar>
            <Nav className="mr-auto" navbar></Nav>
            <NavbarText>
              {this.props.isSignedIn ? (
                <div>
                  <ProfileDropdown
                    onRouteChange={this.props.onRouteChange}
                    toggleModal={this.props.toggleModal}
                    user={user}
                  ></ProfileDropdown>
                </div>
              ) : (
                <div>
                  <Link to="/signin">
                    <button className="button-secondary">Sign In</button>
                  </Link>
                  <Link to="/register">
                    <button className="button-primary" color="secondary">
                      Register
                    </button>
                  </Link>
                </div>
              )}
            </NavbarText>
          </Collapse>
        </Navbar>
      </div>
    );
  }
}

export default NavBar;
