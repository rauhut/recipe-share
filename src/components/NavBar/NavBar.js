import ProfileDropdown from "../Profile/ProfileDropdown";
import React from "react";
import { Link } from "react-router-dom";
import {
  Collapse,
  Navbar,
  Nav,
  Input,
  NavbarText,
  NavbarToggler,
} from "reactstrap";
import "./NavBar.css";

class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isOpen: false,
      searchEntry: "",
      recipes: [],
    };
  }

  onSearchChange = (event) => {
    this.setState({ searchEntry: event.target.value });
  };

  toggle = () => {
    this.setState({ isOpen: !this.state.isOpen });
  };

  onSubmitSearch = () => {
    this.props.onRecipeSearch(this.state.searchEntry);
    this.props.history.push("/search");
  };

  onEnter = (e) => {
    if (e.which === 13) {
      this.onSubmitSearch();
    }
  };

  render() {
    const { user } = this.props;
    const { isOpen } = this.state;

    return (
      <div>
        <Navbar light expand="md">
          <Collapse className="home-nav" isOpen={isOpen} navbar>
            <Link to={"/"}>Recipe Share</Link>
          </Collapse>
          <div className="search" onKeyPress={this.onEnter}>
            <Input
              className="search-bar"
              type="search"
              name="search"
              id="searchInput"
              placeholder="Search for a Recipe"
              onChange={this.onSearchChange}
            />
            <button
              className="button-primary"
              onClick={() => this.onSubmitSearch()}
            >
              Search
            </button>
          </div>
          <NavbarToggler id="nav-toggle" onClick={this.toggle} />
          <Collapse isOpen={isOpen} navbar>
            <Nav className="mr-auto" navbar></Nav>
            <NavbarText>
              {this.props.isSignedIn ? (
                <div>
                  <ProfileDropdown
                    onRouteChange={this.props.onRouteChange}
                    toggleModal={this.props.toggleModal}
                    user={user}
                  ></ProfileDropdown>
                </div>
              ) : (
                <div>
                  <Link to="/signin">
                    <button className="button-secondary">Sign In</button>
                  </Link>
                  <Link to="/register">
                    <button className="button-primary" color="secondary">
                      Register
                    </button>
                  </Link>
                </div>
              )}
            </NavbarText>
          </Collapse>
        </Navbar>
      </div>
    );
  }
}

export default NavBar;
