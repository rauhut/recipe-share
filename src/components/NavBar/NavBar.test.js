import React from "react";
import { shallow } from "enzyme";
import NavBar from "./NavBar";

describe("NavBar", () => {
  let wrapper;

  const mockPropsSignedIn = {
    isSignedIn: true,
    user: {
      id: "123456789",
      username: "testuser",
      recipes: "",
    },
  };

  const mockPropsSignedOut = {
    isSignedIn: false,
    user: {},
  };

  it("renders for signedin user", () => {
    wrapper = shallow(<NavBar {...mockPropsSignedIn} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("renders for signed out user", () => {
    wrapper = shallow(<NavBar {...mockPropsSignedOut} />);
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it("toggles the nav bar on click", () => {
    wrapper.find('[id="nav-toggle"]').simulate("click");
    expect(wrapper.state("isOpen")).toEqual(true);
  });

  it("should update search input state on text change", () => {
    wrapper.find('[id="searchInput"]').simulate("change", {
      target: { value: "test input" },
    });
    expect(wrapper.state("searchEntry")).toEqual("test input");
  });
});
