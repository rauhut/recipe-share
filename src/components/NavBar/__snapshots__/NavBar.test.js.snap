// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders for signed out user 1`] = `
"<div>
  <Navbar light={true} expand=\\"md\\" tag=\\"nav\\">
    <Collapse className=\\"home-nav\\" isOpen={false} navbar={true} in={false} mountOnEnter={false} unmountOnExit={false} appear={false} enter={true} exit={true} onEnter={[Function: noop]} onEntering={[Function: noop]} onEntered={[Function: noop]} onExit={[Function: noop]} onExiting={[Function: noop]} onExited={[Function: noop]} tag=\\"div\\" timeout={350}>
      <Link to=\\"/\\">
        Recipe Share
      </Link>
    </Collapse>
    <div className=\\"search\\" onKeyPress={[Function (anonymous)]}>
      <Input className=\\"search-bar\\" type=\\"search\\" name=\\"search\\" id=\\"searchInput\\" placeholder=\\"Search for a Recipe\\" onChange={[Function (anonymous)]} />
      <button className=\\"button-primary\\" onClick={[Function: onClick]}>
        Search
      </button>
    </div>
    <NavbarToggler id=\\"nav-toggle\\" onClick={[Function (anonymous)]} tag=\\"button\\" type=\\"button\\" />
    <Collapse isOpen={false} navbar={true} in={false} mountOnEnter={false} unmountOnExit={false} appear={false} enter={true} exit={true} onEnter={[Function: noop]} onEntering={[Function: noop]} onEntered={[Function: noop]} onExit={[Function: noop]} onExiting={[Function: noop]} onExited={[Function: noop]} tag=\\"div\\" timeout={350}>
      <Nav className=\\"mr-auto\\" navbar={true} tag=\\"ul\\" vertical={false} />
      <NavbarText tag=\\"span\\">
        <div>
          <Link to=\\"/signin\\">
            <button className=\\"button-secondary\\">
              Sign In
            </button>
          </Link>
          <Link to=\\"/register\\">
            <button className=\\"button-primary\\" color=\\"secondary\\">
              Register
            </button>
          </Link>
        </div>
      </NavbarText>
    </Collapse>
  </Navbar>
</div>"
`;

exports[`NavBar renders for signedin user 1`] = `
"<div>
  <Navbar light={true} expand=\\"md\\" tag=\\"nav\\">
    <Collapse className=\\"home-nav\\" isOpen={false} navbar={true} in={false} mountOnEnter={false} unmountOnExit={false} appear={false} enter={true} exit={true} onEnter={[Function: noop]} onEntering={[Function: noop]} onEntered={[Function: noop]} onExit={[Function: noop]} onExiting={[Function: noop]} onExited={[Function: noop]} tag=\\"div\\" timeout={350}>
      <Link to=\\"/\\">
        Recipe Share
      </Link>
    </Collapse>
    <div className=\\"search\\" onKeyPress={[Function (anonymous)]}>
      <Input className=\\"search-bar\\" type=\\"search\\" name=\\"search\\" id=\\"searchInput\\" placeholder=\\"Search for a Recipe\\" onChange={[Function (anonymous)]} />
      <button className=\\"button-primary\\" onClick={[Function: onClick]}>
        Search
      </button>
    </div>
    <NavbarToggler id=\\"nav-toggle\\" onClick={[Function (anonymous)]} tag=\\"button\\" type=\\"button\\" />
    <Collapse isOpen={false} navbar={true} in={false} mountOnEnter={false} unmountOnExit={false} appear={false} enter={true} exit={true} onEnter={[Function: noop]} onEntering={[Function: noop]} onEntered={[Function: noop]} onExit={[Function: noop]} onExiting={[Function: noop]} onExited={[Function: noop]} tag=\\"div\\" timeout={350}>
      <Nav className=\\"mr-auto\\" navbar={true} tag=\\"ul\\" vertical={false} />
      <NavbarText tag=\\"span\\">
        <div>
          <ProfileDropdown onRouteChange={[undefined]} toggleModal={[undefined]} user={{...}} />
        </div>
      </NavbarText>
    </Collapse>
  </Navbar>
</div>"
`;
